webpack/assets/javascripts/react_app/components/ForemanModal/index.js

```diff
@@ -16,23 +17,25 @@ export const reducers = { foremanModals: reducer };
 const ConnectedForemanModal = props => {
   const { id, title } = props;
   const isOpen = useSelector(state => selectIsModalOpen(state, id));
+  const isSubmitting = useSelector(state => selectIsModalSubmitting(state, id));
   const dispatch = useDispatch();
   const onClose = () => dispatch(setModalClosed({ id }));
-  const modalAlreadyExists = useSelector(state =>
-    selectModalStateById(state, id)
-  );
+  const onLoad = () =>
+    dispatch(addModal({ id, isOpen: false, isSubmitting: false }));
+  const modalExists = useSelector(state => selectModalExists(state, id));
 
   useEffect(() => {
-    if (modalAlreadyExists) return; // don't add modal if it already exists
-    dispatch(addModal({ id, open: false }));
-  }, [modalAlreadyExists, id]); // eslint-disable-line react-hooks/exhaustive-deps
+    if (modalExists) return; // don't add modal if it already exists
+    onLoad();
+  }, [modalExists, id]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
if id has been changed, also `modalExists` changes,  is it needed to add `id` here?

Reply:
I agree `id` doesn't need to be added.  Also I feel we should stop disabling the eslint rule here.